pkg/api/v1/clients/factory/resource_client_factory.go

```diff
@@ -50,10 +59,19 @@ func newResourceClient(factory ResourceClientFactory, params NewResourceClientPa
 		if opts.SharedCache == nil {
 			return nil, errors.Errorf("must provide a shared cache for the kube resource client")
 		}
-		if opts.Cfg == nil {
-			return nil, errors.Errorf("must provide a resclient.Config for the kube resource client")
+
+		// Validate namespace list:
+		// 1. If no namespace list was provided, default to all namespaces
+		// 2. Error if namespace list contains the empty string plus other values
+		namespaces := opts.Namespaces
+		if namespaces == nil || len(namespaces) == 0 {
```

Comment:
nit: `len(namespaces) == 0` covers the `namespaces == nil`

https://golang.org/pkg/builtin/#len